src/Log/Log_Definer.php

```diff
@@ -0,0 +1,29 @@
+<?php declare( strict_types=1 );
+
+namespace Tribe\Libs\Log;
+
+use DI;
+use Monolog\Logger as MonoLogger;
+use Monolog\Handler\StreamHandler;
+use MHCG\Monolog\Handler\WPCLIHandler;
+use Tribe\Libs\Container\Definer_Interface;
+
+class Log_Definer implements Definer_Interface {
+
+	public function define(): array {
+		return [
+			Psr\Log\LoggerInterface::class => DI\factory( function () {
+				$path        = WP_CONTENT_DIR . '/square-one-' . date( 'Y-m-d' ) . '.log';
+				$path        = apply_filters( 'tribe/log/path', $path );
+				$handler     = new StreamHandler( $path, MonoLogger::DEBUG );
+				$cli_handler = new WPCLIHandler( MonoLogger::DEBUG );
+
+				$logger = new Logger( $handler );
+				$logger->add_push_handler( $cli_handler );
```

Comment:
We should only be adding the WP_CLI logger when running in the CLI context. Maybe...

```php
$path        = WP_CONTENT_DIR . '/square-one-' . date( 'Y-m-d' ) . '.log';
$path        = apply_filters( 'tribe/log/path', $path );
$level       = defined( 'LOG_LEVEL' ) ? LOG_LEVEL : MonoLogger::DEBUG;
if ( defined( 'WP_CLI' ) && WP_CLI ) {
  $handler = new WPCLIHandler( $level );
} else {
  $handler = new StreamHandler( $path, $level );
}

return new Logger( $handler );
```

Reply:
I've updated this, let me know what you think. Added another define to enable it for CLI so that you can use it to show the usual messages in a CLI script (replacing `WP_CLI::line/error` etc...) but it will also simultaneously log the same message to a file.

Reply:
I like it.